### ISSUE TITLE
Running JDBC instrumentation causing long socket read times from DB 

### ISSUE THREAD

**Issue by PaarthB** [open-telemetry/opentelemetry-java#5386]
**Describe the bug**
Running JDBC instrumentation causing delays in socket read from DB . This causes application threads to wait on DB, causing thread pool (Tomcat) starvation. This appears to be some sort of overhead introduced by the JDBC instrumentation running.

**Steps to reproduce**
Use the JDBC instrumentation in Java OpenTelemery agent and observe load on DB CPU and also Write IOPs and application thread running times

**What did you expect to see?**
The service runs as normal without any potential issues causing DB CPU to go up more than expected, or application threads (Tomcat) to start waiting more than often on socket read from DB.

**What did you see instead?**
The service started waiting on response from DB longer than usual, causing threadpool starvation in the web service.

**What version and what artifacts are you using?**
Artifacts: (`opentelemetry-bom`, `opentelemetry-javaagent`, `opentelemetry-sdk` , `opentelemetry-api` which exporters, etc)
Version: (`v1.21.0`)
How did you reference these artifacts? (`pom.xml`)

**Environment**
Compiler: "Java 11 temurin11"
OS: "Debian 11.6" 
Runtime (if different from JDK above):
OS (if different from OS compiled on):

**Additional context**
To confirm this, we found that the issue went away once we turned the JDBC instrumentation off.
The name of the instrumentation is `otel.instrumentation.jdbc.enabled` which is set to `false` when running the JAR. This can also be confirmed here: https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#suppressing-specific-agent-instrumentation

Previous issue created: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8303

**Comment by PaarthB** [comment]
It also appears, the most waiting is done by the operation from `io​.opentelemetry​.javaagent​.shaded​.instrumentation​.jdbc​.internal​.Db​Request​.create`

**Comment by trask** [comment]
closing in favor of https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8303
